IoTHubBuddy/IoTViewer/IoTDataManager.cs

```diff
@@ -0,0 +1,228 @@
+﻿using IoTHubBuddy.Models;
+using Microsoft.Azure.Devices;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Data.Json;
+
+namespace IoTHubBuddy
+{
+    static class IoTDataManager
+    {
+        private const string AzureResourceUri = "https://management.azure.com/";
+
+        /// <summary>
+        /// check validity for returned json object. If it doesn't contain a value, then there was an error in gathering the data
+        /// Error parsing/handling will be implemented later
+        /// </summary>
+        /// <param name="jo"></param>
+        /// <returns></returns>
+        private static bool checkJsonObject(JsonObject jo)
+        {
+            if (!jo.ContainsKey("value"))
+            {
+                return false;
+            }
+            return true;
+        }
+        /// <summary>
+        /// sends a GET resquest to azure management REST API
+        /// </summary>
+        /// <param name="token"></param>
+        /// <param name="relative"></param>
+        /// <returns></returns>
+        public static async Task<JsonObject> GetIoTData(string token, string relative)
+        {
+            if(token == null)
+            {
+                return null;
+            }
+            using(var client = new HttpClient())
+            {
+                client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", token);
+                client.DefaultRequestHeaders.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("application/json"));
+                var restApi = new Uri(AzureResourceUri + relative);
+                var infoResult = await client.GetAsync(restApi);
+                string content = await infoResult.Content.ReadAsStringAsync();
+                var jsonObject = JsonObject.Parse(content);
+                if(checkJsonObject(jsonObject))
+                {
+                   return jsonObject;
+                } else
+                {
+                    return null;
```

Comment:
I will stop complaining about nulls now, but please audit the code for `return null` and see if you can avoid it.